src/components/flame-graph/Canvas.js

```diff
@@ -58,6 +58,38 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
     (this: any)._hitTest = this._hitTest.bind(this);
   }
 
+  getColor(implementation: Implementation, selfTimeRelative: number): string {
```

Comment:
Could you please unit test this function? I think you can break it out into its own function, as it's a pure function. You can still keep it in this file, I don't think you need to break it out into utils or anything.